napari/layers/_shapes_layer/shape_util.py

```diff
@@ -678,3 +678,91 @@ def segment_normal(a, b):
     else:
         unit_norm = normal/norm
     return unit_norm
+
+
+def poly_to_mask(mask_shape, vertices):
+    """Converts a polygon to a boolean mask with `True` for points
+    lying inside the shape. Uses the bounding box of the vertices to reduce
+    computation time.
+
+    Parameters
+    ----------
+    mask_shape : np.ndarray | tuple
+        1x2 array of shape of mask to be generated.
+    vertices : np.ndarray
+        Nx2 array of the vertices of the polygon.
+
+    Returns
+    ----------
+    mask : np.ndarray
+        Boolean array with `True` for points inside the polygon
+    """
+    mask = np.zeros(mask_shape, dtype=bool)
+    bottom = vertices.min(axis=0).astype('int')
+    top = np.ceil(vertices.max(axis=0)).astype('int')
+    top = np.append([top], [mask_shape], axis=0).min(axis=0)
+    if np.all(top > bottom):
+        bb_mask = grid_points_in_poly(top - bottom, vertices - bottom)
+        mask[bottom[0]:top[0], bottom[1]:top[1]] = bb_mask
+    return mask
+
+
+def grid_points_in_poly(shape, vertices):
+    """Converts a polygon to a boolean mask with `True` for points
+    lying inside the shape. Loops through all indices in the grid
+
+    Parameters
+    ----------
+    shape : np.ndarray | tuple
+        1x2 array of shape of mask to be generated.
+    vertices : np.ndarray
+        Nx2 array of the vertices of the polygon.
+
+    Returns
+    ----------
+    mask : np.ndarray
+        Boolean array with `True` for points inside the polygon
+    """
+    points = np.array([(x, y) for x in range(shape[0])
+                      for y in range(shape[1])], dtype=int)
+    inside = points_in_poly(points, vertices)
+    mask = inside.reshape(shape)
+    return mask
+
+
+def points_in_poly(points, vertices):
+    """Tests points for being inside a polygon.
+
+    Parameters
+    ----------
+    points : np.ndarray
+        Mx2 array of points to be tested
+    vertices : np.ndarray
+        Nx2 array of the vertices of the polygon.
+
+    Returns
+    ----------
+    inside : np.ndarray
+        Length M boolean array with `True` for points inside the polygon
+    """
+    n_verts = len(vertices)
+    inside = np.zeros(len(points), dtype=bool)
+    j = n_verts-1
+    for i in range(n_verts):
+        cond_1 = np.logical_and(vertices[i, 1] <= points[:, 1],
```

Comment:
could you provide comments as to what each condition means?

Reply:
I didn't comment every condition but put the name of the algorithm in the doc string (it is the ray casting algorithm) and I put some comments about what is being tested in each block, and why those tests then lead to the inside / outside determination. Hopefully this helps